### ISSUE TITLE
Provider forces use of biometric unlock, even when desktop app is not installed

### ISSUE THREAD

**Issue by YenTheFirst** [1Password/terraform-provider-onepassword#204]
### Your environment

Terraform Provider Version: 2.1.2

Connect Server Version: N/A

CLI Version: 2.24.0

OS: Linux, ubuntu 24.04, kernel 6.8.0-47

Terraform Version: 1.5.7

## What happened?
I configured a simple data source, and executed `terraform plan`.
It hung for a few seconds, then failed with the error
```
Unable to read vault, got error: op error: connecting to desktop app: cannot connect to 1Password app, make sure it is running
```

## What did you expect to happen?
I expected it to either prompt for a password, or, to use the results of a recent `op signin`, and fetch the data.

## Steps to reproduce
1. Install the `op` cli, but not the desktop app.
2. Configure a basic data resource in terraform
```
terraform {
  required_providers {
    onepassword = {
      source = "1Password/onepassword"
      version = "2.1.2"
    }
  }
}

provider "onepassword" {
  account = "https://myaccount.1password.com"
}

data "onepassword_vault" "the-vault" {
  name = "the-vault"
}

data "onepassword_item" "relevant-item" {
  vault = data.onepassword_vault.the-vault.uuid
  title   = "Important Item"
}
```
3. disable biometric unlock, login to `op`, and terraform plan.
```
$ export OP_BIOMETRIC_UNLOCK_ENABLED=false
$ eval $(op signin)
$ terraform plan
```
4. Observe the error
```
╷
│ Error: Client Error
│ 
│   with data.onepassword_vault.the-vault,
│   on test.tf line 36, in data "onepassword_vault" "the-vault":
│   36: data "onepassword_vault" "the-vault" {
│ 
│ Unable to read vault, got error: op error: connecting to desktop app: cannot connect to 1Password app, make sure it
│ is running
╵
Releasing state lock. This may take a few moments...
```

### Notes

Debugging this with `strace`, I noticed that `OP_BIOMETRIC_UNLOCK_ENABLED` was being overriden to true, regardless of whether I'd specified otherwise in the outer environment.

It looks like this is currently encoded here https://github.com/1Password/terraform-provider-onepassword/blob/b942ecfa7ddabb48916af46b62369f8d454ebd28/internal/onepassword/cli/op.go#L258, to always set `OP_BIOMETRIC_UNLOCK_ENABLED=true` when using an account URL, instead of connect or a service token.

however, that behavior doesn't seem to be documented, and it's quite inconvenient. The relevant PR (https://github.com/1Password/terraform-provider-onepassword/pull/167) doesn't seem to say why it was added - though I'm unaware if the history goes back further.

**Comment by framigni** [comment]
Indeed, confirming the bug here. It's very inconvenient, because my station does not have biometrics and I'd rather much prefer not having the desktop app

**Comment by volodymyrZotov** [comment]
We released a new provider version `v3.x` that doesn't use 1Password CLI under the hood anymore. For biometric unlock it uses [1password sdk](https://developer.1password.com/docs/sdks/).

As there is no CLI used by the provider anymore we do not plan to investigate this issue.

Please update provider to `v3`.
